components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java

```diff
@@ -6110,6 +6117,67 @@ public String getMainAppId(String sharedAppId) throws IdentityApplicationManagem
         }
     }
 
+    @Override
+    public String getOwnerOrgId(String sharedAppId) throws IdentityApplicationManagementServerException {
+
+        try (Connection connection = IdentityDatabaseUtil.getDBConnection(false)) {
+            try (NamedPreparedStatement namedPreparedStatement = new NamedPreparedStatement(
+                    connection, ApplicationMgtDBQueries.GET_OWNER_ORG_ID_BY_SHARED_APP_ID)) {
+                namedPreparedStatement.setString(DB_SCHEMA_COLUMN_NAME_SHARED_APP_ID, sharedAppId);
+                try (ResultSet resultSet = namedPreparedStatement.executeQuery()) {
+                    if (resultSet.next()) {
+                        return resultSet.getString(DB_SCHEMA_COLUMN_NAME_OWNER_ORG_ID);
+                    }
+                }
+            }
+            return null;
+        } catch (SQLException e) {
+            throw new IdentityApplicationManagementServerException(
+                    String.format("Error while getting owner organization id for the shared application with id: %s",
+                            sharedAppId), e);
+        }
+    }
+
+    @Override
+    public Map<String, String> getSharedApplicationIds(String mainAppId, String ownerOrgId, List<String> sharedOrgIds)
+            throws IdentityApplicationManagementServerException {
+
+        if (CollectionUtils.isEmpty(sharedOrgIds)) {
+            return Collections.emptyMap();
+        }
+
+        Map<String, String> sharedAppIds = new HashMap<>();
+
+        String placeholders = IntStream.range(0, sharedOrgIds.size())
+                .mapToObj(i -> ":" + SHARED_ORG_ID_PLACEHOLDER_PREFIX + i + ";")
+                .collect(Collectors.joining(", "));
+        String sqlStmt = GET_FILTERED_SHARED_APPLICATIONS.replace(SHARED_ORG_ID_LIST_PLACEHOLDER, placeholders);
+
+        try (Connection connection = IdentityDatabaseUtil.getDBConnection(false)) {
+
+            try (NamedPreparedStatement namedPreparedStatement =
```

Comment:
add both resources into the same try block

Reply:
Addressed with https://github.com/wso2/carbon-identity-framework/pull/6129/commits/1c6e52945f25edf3ca8faf6da8d114cf1439aed3